enhancements/update/reduced_reboots.md

```diff
@@ -0,0 +1,328 @@
+that 4.7 to 4.8.z and 4.8.z-n to 4.8.z upgrades continue to work as well.
+
+Teams which believe this is not achievable or the level of effort is extremely
+high should document those findings.
+
+Thus far RHCOS, Node, MCO, SDN, Containers, and PSAP teams are known to fall into
+this group of teams which have components coupled to the host.
+
+#### MCO - Widen Node Constraints to allow for N-2
+
+Building upon the EUS-to-EUS upgrade MVP work(https://github.com/openshift/enhancements/blob/master/enhancements/update/eus-upgrades-mvp.md#mco---enforce-openshifts-defined-host-component-version-skew-policies)
+to allow MCO to enforce host constraints we will broaden those constraints to
+enable host component version skew.
+
+Admins who choose to would then be able to skip a host reboot by following this
+pattern:
+
+1. Starting with a 4.8 cluster, pause Worker MachineConfigPool
+1. Upgrade to 4.9
+1. Upgrade to 4.10
+1. Unpause Worker MachineConfigPool
+
+Note that this can be decoupled in a way that when we ship 4.9 the initial MCO
+could assert constraints which require 4.9 host components before upgrading to
+4.10. Then after we ship 4.10 and have sufficiently tested a 4.8 to 4.10 host
+component version skew a later 4.9.z MCO could have its constraints broadened.
+This allows us additional time to test broader version skews if we so choose.
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+### Risks and Mitigations
+
+This imposes significant risk due to a number of factors:
+- We're currently not confident in upstream's testing matrix and our specific
+  feature sets
+- We've never before expected teams to offer broader than N-1 compatibility.
+  Teams have always assumed at most N-1 and even then, especially early after a
+  GA release, it's not uncommon to find problems in N-1 compatibility.
+- While N-1 is tested in the context of upgrades it's not been tested in long
+  term use.
+- If external integrations depend on host components of all nodes having been
+  updated then we'll run into problems. For instance, if there's an upgrade
+  scenario where RHV cloud provider integration needs to be upgraded between 4.6
+  and 4.10 in order to ensure compatibility and the components which interface
+  with RHV are components of RHCOS then we may not upgrade those components at
+  the same minor version expected previously.
+
+We may mitigate some of this by further delaying EUS-to-EUS upgades after normal
+minor version upgrades have been promoted to stable and allocating significantly
+more time and effort to testing efforts. Ultimately this introduces another
+dimension to an already complex testing matrix.
+
+## Design Details
+
+### Open Questions [optional]
```

Comment:
To prevent an upgrade of specific nodes, the high-value assets of a cluster with special resources and constraints (kernel, os, ... ), one could create a custom MCP and set either `pause: true` or `maxUnavailable: 0` this would prevent MCO from updating this very MCP for the next 1 or 2 release depending on the constraint. 

An operator that does preflight checks on newer kernel, os could use the new information that is rolled out to the other nodes to check if the special resource would work on the update.

Reply:
I think if an operator has tigheter constraints than can be expressed by either minimum version of MachineConfig, kubelet version, or kernel then that logic should live outside of the MCO. The MCO is caring about this only because it can affect the host component versions.

Reply:
@sdodson Understood, this is related to this: https://github.com/openshift/enhancements/pull/357 my questions around this is, how SRO can prevent upgrades and guard the special resources from unwanted upgrades. Some kmods are sensitive to any kernel version change and some kmods (kABI whitelisted symbols) only care of about OS major changes (8.x -> 9.x) .